scripts/check_test_modules.sh

```
@@ -0,0 +1,88 @@
+# not a compile error — it is simply never compiled and never run, and the suite
+# stays green while the coverage is gone.
+#
+# The per-binary runner this replaced refused to run when the binary count did
+# not match the file count. This is that guard. It is wired into BOTH `make
+# check` (local + pre-commit hook) and the CI `check` job directly, because CI
+# invokes the cargo commands itself rather than going through `make check` —
+# wiring it only into the Makefile would leave CI unguarded.
+
+set -uo pipefail
+
+cd "$(git rev-parse --show-toplevel)"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)